### ISSUE TITLE
[Waiting on checklist] [$250] Expensify Card - Verifying page does not show offline indication when user is offline

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#51885]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.56-3
**Reproducible in staging?:** Y
**Reproducible in production?:** Unable to check
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):**applausetester+khcategory18@applause.expensifail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Go to staging.new.expensify.com
2. Go to workspace settings > Expensify Card
3. Click Issue new card
4. Add a bank account if not yet
5. Go offline
6. Select the added bank account

## Expected Result:
Verifying page will show that user is offline

## Actual Result:
Verifying page shows the animation as long as user is offline

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [x] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [x] iOS: HybridApp
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/34fb3f04-9278-4a10-9d47-667c547c759e

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021852431360082320267</li>
        <li>Upwork Job ID: 1852431360082320267</li>
        <li>Last Price Increase: 2024-11-01</li>
<li>Automatic offers: </li>
<ul>
<li>ahmedGaber93 | Reviewer | 104729929</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @ahmedGaber93</details>

**Comment by lanitochka17** [comment]
We think that this bug might be related to #wave-collect - Release 2

**Comment by francoisl** [comment]
Sounds like it could come from https://github.com/Expensify/App/pull/51407.
The proposal makes sense to me, going to label External to get a second pair of eyes on it.

**Comment by ahmedGaber93** [comment]
reviewing...

**Comment by ahmedGaber93** [comment]
@Nodebrute Thanks for the proposal

Your solution not work with me, unless I remove `!shouldUseNarrowLayout && ` from the below condition. 
https://github.com/Expensify/App/blob/20fadd74acab837fa9a89c6bf5c71879df118601/src/components/ScreenWrapper.tsx#L298

I think your RCA is not complete.

**Comment by ahmedGaber93** [comment]
@francoisl What do you think about the expected behavior here
1. display `OfflineIndicator` at the bottom of screen
2. display `FullPageOfflineBlockingView`

**Comment by ahmedGaber93** [comment]
> for wide screen we never show the offline indicator on the RHP

Hmm! I am not sure about that, Let's get confirmation from @Expensify/design for https://github.com/Expensify/App/issues/51885#issuecomment-2452602748

**Comment by dubielzyk-expensify** [comment]
I think this one is the right one (as mentioned in [this proposal](https://github.com/Expensify/App/issues/51885#issuecomment-2452574878)):
![CleanShot 2024-11-04 at 09 20 27@2x](https://github.com/user-attachments/assets/abc8eae2-d51b-44a4-8bbe-bcba5d500104)

**Comment by dubielzyk-expensify** [comment]
I think given that this whole flow happens in the right hand panel and are step by step flows, I think that pattern is the most right cause you can't proceed without being online right? The bottom subtle offline indicator is more appropriate when you're on a screen but you can still do something useful.

**Comment by koko57** [comment]
@dubielzyk-expensify @francoisl as I was working on the PR that caused this issue, I would like to work on that one

**Comment by dubielzyk-expensify** [comment]
I don't think it hurts to add it to bottom of this screen either (as that's what we usually do):
![CleanShot 2024-11-04 at 14 31 21@2x](https://github.com/user-attachments/assets/007252f9-dae7-46e5-aee1-cf4273420135)

I'll see if @Expensify/design has other opinions :)

**Comment by shawnborton** [comment]
That makes total sense to me, and feels consistent with other parts of the product as you mention.

**Comment by Beamanator** [comment]
Assigning @koko57 since you mentioned you worked on the PR that caused this issue 🙏 

Do we all understand the desired solution? If so, @koko57 it would be great if you could get a PR submitted soon since this is a deploy blocker! 🙏

**Comment by koko57** [comment]
thanks @Beamanator!

so I think we should go with this - https://github.com/Expensify/App/issues/51885#issuecomment-2453615956 this way we won't be able to choose the bank account, so we won't be showing the loading/verifying/success states

cc @dubielzyk-expensify do you agree?

**Comment by Beamanator** [comment]
> @Beamanator am i eligible for compensation here ? given that help wanted label was attached here and my solution would be used to fix the bug ?

@twisterdotcom can confirm, but yes @twilight2294 you should be eligible for compensation if your solution is used for the fix 👍

**Comment by Julesssss** [comment]
I'm removing this as a deploy blocker. It doesn't block a flow, is an edge case, and, most importantly, also exists in production.

**Comment by koko57** [comment]
https://github.com/Expensify/App/issues/51885#issuecomment-2454383027 EDIT: but yeah, if the Bank account is ALREADY in the verifying/verified state we should show the modals

**Comment by koko57** [comment]
yes, it's exactly what I think it's ok to do here

**Comment by koko57** [comment]
@dubielzyk-expensify @shawnborton asking for confirmation, as I'm completing the checklist in the meantime

we'll be displaying the Verifying/Verified state when already in this state
<img width="1512" alt="Screenshot 2024-11-04 at 18 21 55" src="https://github.com/user-attachments/assets/96fcc9db-44b3-4e34-a709-41f3b68f3b34">

but we'll display the offline indicator for the page that we need to choose the bank account 
<img width="1506" alt="Screenshot 2024-11-04 at 18 12 11" src="https://github.com/user-attachments/assets/cd0f0e5a-9141-40d2-aae9-e3ea7496afa3">

**Comment by shawnborton** [comment]
I think that sounds right to me, but will let others chime in as well.

**Comment by twisterdotcom** [comment]
Agree here. @twilight2294 happy to pay for your solution, but as you aren't doing the work, it will be $125 for you, not the full $250.

**Comment by ahmedGaber93** [comment]
# BugZero Checklist:

- [x] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [x] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [ ] 2a. Reported on production
  - [x] 2b. Reported on staging (deploy blocker)
  - [ ] 2c. Reported on both staging and production
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [x] 3z. Other: Applause - Internal Team

</details>

- [x] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment: https://github.com/Expensify/App/pull/51407/files#r1843618525

- [x] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion: N/A

- [x] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.




- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:




### Test:


1. Create new Workspace
2. Go to Settings > Workspace > More features
3. Enable Expensify Card
4. Go offline
5. Go to Expensify Card page, click Issue New card
6. Verify the Choose Bank Account page should display the full page oflline blocking view.
7. Try the same with workspaces that already have the Bank account verified or in verifying state - you should see the modal with this state instead


Do we agree 👍 or 👎
